compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -129,6 +132,15 @@ unsafe fn configure_llvm(sess: &Session) {
 
     llvm::LLVMInitializePasses();
 
+    for plugin in &sess.opts.debugging_opts.llvm_plugins {
+        let path = Path::new(plugin);
+        let res = DynamicLibrary::open(path);
```

Comment:
I think you want to `mem::forget(res)` here, otherwise the library will get almost immediately closed in `Drop` hook of `DynamicLibrary`.

Reply:
Adding mem::forget makes sense since we want the plugin to be available during the entire compilation process.
However, for my testcase it has been also working before. Is this expected (probably because we have a small test program and the compilation finished before the dropping takes effect (?)), or is this an issue?

Reply:
Often dlclose does nothing, but you absolutely should not rely on it doing nothing.